### PR TITLE
Added the ability to specify a Spin Count Unit via a GC Configuration to release/7.0

### PR DESCRIPTION
**Pull request by mrsharm**
Added the ability to specify a Spin Count Unit via a GC Configuration for the release/7.0-staging branch and make use of this value in the SetYieldProcessorScalingFactor function if the value is valid. If this configuration is not specified, we default to 0 and fall back to the original logic.

This PR is port of https://github.com/dotnet/runtime/pull/84339 as automatic merging failed. 

## Customer impact
There were performance regressions experienced by customers related to [this](https://github.com/dotnet/runtime/pull/68881) fix that resulted in an increase in the Spin Count Unit. This PR allows the ability to hardcode that value. 

## Testing
Tested manually and verified by a customer that the fix reverted the regression.

## Risk
Low. The setting is not enabled by default and has to be explicitly set by the user for the conditional logic to work.

### CODE REVIEW

**Comment by teo-tsirpanis** [comment]
@mrsharm you have to target the `release/7.0-staging` branch.

**Comment by JulieLeeMSFT** [comment]
@mrsharm, I am removing the servicing consider label until it is reviewed and all the tests are passed.

File: src/coreclr/gc/gc.cpp
Comment (Maoni0) [inline src/coreclr/gc/gc.cpp:45687]: you are missing part of the changes in `GCHeap::SetYieldProcessorScalingFactor` (replace the integer with `MAX_YP_SPIN_COUNT_UNIT`)

**Comment by carlossanlop** [comment]
@mrsharm today is code complete. If you want this fix included in the May Servicing release, please get it ready by 4pm PT today:

- [ ] Get the Tactics approval ASAP
- [ ] Confirm the CI failures are unrelated
- [ ] Get a sign-off from another area owner

**Comment by Maoni0** [comment]
both @mrsharm and @mangod9 are out of office and I was not informed this needed to be in for the next servicing release when we were going through issues last week. so I will assume this does not need to get in right now and let them handle it when they are back.

**Comment by JulieLeeMSFT** [comment]
@mrsharm, please update the issue following the template for servicing fix (refer https://github.com/dotnet/runtime/pull/84444).

**Review by Maoni0 — APPROVED** [review]

**Comment by mangod9** [comment]
Correct, we are still waiting for customer validation on this and will merge when we have confirmation.

**Review by jeffschwMSFT — APPROVED** [review]
approved. we will bring for consideration in 7.0.x

**Pull request merged**
